src/impl_methods.rs

```diff
@@ -321,9 +321,9 @@ where
     ///
     /// **Panics** if an index is out of bounds or step size is zero.<br>
     /// (**Panics** if `D` is `IxDyn` and `info` does not match the number of array axes.)
-    pub fn slice_mut<Do>(&mut self, info: &SliceInfo<D::SliceArg, Do>) -> ArrayViewMut<A, Do>
+    pub fn slice_mut<I>(&mut self, info: &I) -> ArrayViewMut<A, I::OutDim>
```

Comment:
impl Trait would be ideal here as a user interface, but it seems like we can't access the associated type in any way then.

Reply:
Yeah, that's the issue. We could use `impl CanSlice<D>` for `slice_collapse`/`slice_inplace`, though. IMO the `I` param is nice for consistency with the other slicing methods, but I'd be fine with `impl CanSlice<D>` too.